docs/src/5-prestashop-billing/4-how-tos/4-handle-plan-selection/README.md

```diff
@@ -14,273 +14,290 @@ You should add your plan presentation in the configuration page for your module
+        ]);
 
-3. Implement `views/js/configure.js` to make billing works with the component instead of the `initialize` method
-
-    ```javascript
-      window?.psaccountsVue?.init();
-
-      let billingContext = { ...window.psBillingContext.context }
-      let currentModal;
-      let customer;
-
-      if(window.psaccountsVue.isOnboardingCompleted() == true) {
-
-        customer = new window.psBilling.CustomerComponent({
-          context: billingContext,
-          hideInvoiceList: true,
-          onOpenModal,
-          onEventHook
-        });
-        customer.render('#ps-billing');
+        // ...
       }
+   ```
 
-      // Modal open / close management
-      async function onCloseModal(data) {
-        await Promise.all([currentModal.close(), updateCustomerProps(data)]);
-      };
-
-      function onOpenModal(type, data) {
-        currentModal = new window.psBilling.ModalContainerComponent({
-          type,
-          context: {
-            ...billingContext,
-            ...data,
-          },
-          onCloseModal,
-          onEventHook
-        });
-        currentModal.render('#ps-modal');
-      };
-
-      function updateCustomerProps(data) {
-        return customer.updateProps({
-          context: {
-            ...billingContext,
-            ...data,
-          },
-        });
-      };
-
-      function onEventHook(type, data) {
-        // Do anything
-      }
-    ```
-
-
-4. Display your plan in your module and hide the subscription management in `views/templates/admin/configure.tpl`
+3. Retrieve plan pricing from Billing API and inject it into your template
 
-    ```html{4-26, 28, 30}
+   ```php{43-56,77-78}
+        // ...
+        if (!empty($productComponents['body']['items']))
+        {
+          // You need to retrieve plan information from your API or class of array like this
+          // For plan id, please get in touch with your Solution Engineer at PrestaShop.
+          $planInfos = [
+              [
+                  "id" => "<module_name>_free",
+                  "name" => '<module_name> Free',
+              ],
+              [
+                  "id" => "<module_name>_advanced",
+                  "name" => '<module_name> Advanced',
+              ],
+              [
+                  "id" => "<module_name>_ultimate",
+                  "name" => '<module_name> Ultimate',
+              ]
+          ];
+
+          // Sorts you plans informations and plans from Billing API
+          // to be sure to have the same order
+          array_multisort(array_column($planInfos, 'id'), SORT_ASC, $planInfos);
+          array_multisort(array_column($componentItems, 'id') , SORT_ASC, $componentItems);
+
+          // Merge plans from Billing API with your plans informations
+          $componentItems =  array_replace_recursive($planInfos, $componentItems);
+        }
+        // ...
+   ```
+
+4. Inject this file as a script in `views/templates/admin/configure.tpl`
```

Comment:
**issue**: j'ai eu du mal à comprendre de quel "file" on parlait (bon pas aider par le fait qu'on est dans github avec le diff).

Jusqu'ici on parle du fichier `<nom_module>.php` et puis là le "this file" semble correspondre à `configure.js`


Peut-etre que ce serait mieux formulé : "Inject `views/js/configure.js` as a script in `views/templates/admin/configure.tpl`".

Dans ce cas là il faudrait reprendre l'énoncé du 1. https://github.com/PrestaShopCorp/docs.cloud.prestashop.com/pull/69/files#diff-3dbf766a11d32c4bcf20242f82830ddda2c63b8a4c93d0aeeebd1deca3fb08efR17

"1. Create a JS file in your module: `views/js/configure.js` and make it available for your tpl from `<module_name>.php` file." (ou un truc du style)

Reply:
Effectivement, j'ai juste décalé c'était le 2 donc j'ai mis `views/js/configure.js` à là la place de `this file`